core/utils.ts

```diff
@@ -184,6 +184,18 @@ export function validate(compiledGraph: dataform.ICompiledGraph): dataform.IGrap
       validationErrors.push(dataform.ValidationError.create({ message, actionName }));
     }
 
+    // sqlserver config
+    if(!!action.sqldatawarehouse) {
+      if(action.sqldatawarehouse.distribution){
+        let dist = action.sqldatawarehouse.distribution.toUpperCase();
+        let hash_pattern = new RegExp('HASH\\s*\\(\\s*\\w*\\s*\\)\\s*');
```

Comment:
`hashPattern`, and can you make this a constant defined somewhere nearish the top of this file?